jaggr-core/src/main/java/com/ibm/jaggr/core/impl/layer/CacheEntry.java

```diff
@@ -75,39 +83,76 @@
 	 *
 	 * @param request
 	 *            the request object
+	 * @param sourceMapResult
+	 *            (Output) mutable object reference to the source map.  May be null
+	 *            if source maps are not being requested.
 	 * @return The InputStream for the built layer
 	 * @throws IOException
 	 */
-	public InputStream getInputStream(HttpServletRequest request) throws IOException {
-		// Check bytes before filename when reading and reverse order when setting
+	public InputStream getInputStream(HttpServletRequest request, MutableObject<byte[]> sourceMapResult) throws IOException {
+		// Check bytes before filename when reading and reverse order when setting.
+		// The following local variables intentionally hide the instance variables.
 		byte[] bytes = this.bytes;
+		byte[] sourceMap = this.sourceMap;
 		String filename = this.filename;
-		InputStream in = null;
+
+		InputStream result = null;
 		if (bytes != null) {
-			in = new ByteArrayInputStream(bytes);
+			// Cache data is already in memory.  Don't need to de-serialize it.
+			result = new ByteArrayInputStream(bytes);
+			if (sourceMapResult != null && sourceMapSize > 0) {
+				sourceMapResult.setValue(sourceMap);
+			}
 		} else if (filename != null){
+			// De-serialize data from cache
 			ICacheManager cmgr = ((IAggregator)request.getAttribute(IAggregator.AGGREGATOR_REQATTRNAME)).getCacheManager();
 			File file = new File(cmgr.getCacheDir(), filename);
-			in = new FileInputStream(file);
+			if (sourceMapSize == 0) {
+				// No source map data in cache entry so just stream the file.
+				result = new FileInputStream(file);
+			} else {
+				// Entry contains source map data so that means it's a serialized CacheData
+				// instance.  De-serialize the object and extract the data.
+				CacheData data;
+				ObjectInputStream is = new ObjectInputStream(

```

Comment:
I guess so.  Commons lang has SerializationUtils.  The [deserialize](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/SerializationUtils.html#deserialize%28java.io.InputStream%29) method will close the stream, eliminating the need for a finally clause in the caller, and throws [SerializationException](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/SerializationException.html) (a runtime exception) potentially eliminating the need for catch clauses in the caller if you're only catching exceptions to convert them to checked or runtime exceptions.

I'll create another pull request for these changes.